Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2434 @@
+    For this implementation, no special attributes defined
+    """
+    def __init__(self, base_url: str, verify: bool, proxy: bool, client_id: str, client_secret: str,
+                 first_fetch: str = '3 days', fetch_limit: int = 50, is_incident_event: bool = False,
+                 is_fetch_comment: bool = False, fetch_status: list = None, fetch_priority: list = None):
+        super().__init__(
+            base_url=base_url,
+            verify=verify,
+            proxy=proxy
+        )
+
+        self.token_url = f'{base_url}{TOKEN_ENDPOINT}'
+        self.client_key = client_id
+        self.secret_key = client_secret
+        self.first_fetch = first_fetch
+        self.fetch_limit = fetch_limit
+        self.is_incident_event = is_incident_event
+        self.is_fetch_comment = is_fetch_comment
+        self.fetch_status = fetch_status
+        self.fetch_priority = fetch_priority
+        self.sid = None
+
+    def get_access_token(self):
+        """
+        Generate Access token
+        Returns:
+            Returns the access_token
+        """
+        payload = {
+            "grant_type": 'client_credentials'
+        }
+        response = None
+        if last_sid := self.get_sid_from_context():
+            demisto.debug(f"Last login sid still alive. Return last sid {last_sid}")
+            return last_sid
+        else:
+            try:
+                response = self._http_request(
+                    method='POST',
+                    full_url=self.token_url,
+                    auth=(self.client_key, self.secret_key),
+                    data=payload,
+                    error_handler=handle_errors
+                )
+                sid = response.get('access_token', '')
+                self.sid = sid
+                timestamp_string = int(time.time() * 1000)
+                demisto.debug(f"login: success, saving sid={sid} , sid_last_datetime={timestamp_string} to integrationContext")
+                demisto.setIntegrationContext({'sedr_sid': sid, 'sedr_last_sid_datatime': timestamp_string})
+                return sid
+            except Exception as e:
+                msg = f'Something went wrong with request, {e}'
+                demisto.debug(msg)
+                raise DemistoException(msg, res=requests.Response)
+
+    def get_sid_from_context(self):
+        if sid := demisto.getIntegrationContext().get('sedr_sid', None):
+            last_datetime = demisto.getIntegrationContext().get('sedr_last_sid_datatime', None)
+            now_datetime = int(time.time() * 1000)  # Convert to Millisecond
+            time_diff = int(now_datetime - last_datetime)
+            demisto.debug(f"Check Last Sid: {sid}, time_diff: {time_diff}.")
+            return sid if time_diff <= SESSION_TIMEOUT_SEC * 1000 else None
+
+        demisto.debug("No last SID found in Context.")
+        return None
+
+    def query_request_api(self, endpoint: str, params: dict, method: str | None = 'POST') \
```

Comment:
why is method optional?

Reply:
Changes the function arguments. Kind review